From decaf891e5f1406cc80279fb19874125126f5cce Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:10 -0400
Subject: [net] netfilter: NETFILTER_XT_TARGET_LOG selects NF_LOG_*

Message-id: <20141021162110.22888.86443.stgit@localhost.localdomain>
Patchwork-id: 98014
O-Subject: [RHEL7 net PATCH 117/124] netfilter: NETFILTER_XT_TARGET_LOG selects NF_LOG_*
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: d79a61d646db950b68dd79ecc627cb5f11e0d8ac

 commit d79a61d646db950b68dd79ecc627cb5f11e0d8ac
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Fri Aug 29 09:34:49 2014 +0200

    netfilter: NETFILTER_XT_TARGET_LOG selects NF_LOG_*

    CONFIG_NETFILTER_XT_TARGET_LOG is not selected anymore when jumping
    from 3.16 to 3.17-rc1 if you don't set on the new NF_LOG_IPV4 and
    NF_LOG_IPV6 switches.

    Change this to select the three new symbols NF_LOG_COMMON, NF_LOG_IPV4
    and NF_LOG_IPV6 instead, so NETFILTER_XT_TARGET_LOG remains enabled
    when moving from old to new kernels.

    Reported-by: Rafal Milecki <zajec5@gmail.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index f6cce87..349f233 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -756,7 +756,9 @@ config NETFILTER_XT_TARGET_LED
 
 config NETFILTER_XT_TARGET_LOG
  tristate "LOG target support"
- depends on NF_LOG_IPV4 && NF_LOG_IPV6
+ select NF_LOG_COMMON
+ select NF_LOG_IPV4
+ select NF_LOG_IPV6 if IPV6
  default m if NETFILTER_ADVANCED=n
  help
    This option adds a `LOG' target, which allows you to create rules in
-- 
1.7.1